packages/cli/src/ui/hooks/useCompletion.ts

```diff
@@ -530,15 +588,17 @@ export function useCompletion(
           }
         }
       }
-      if (isMounted) {
+      if (isMounted && !abortController.signal.aborted) {
         setIsLoadingSuggestions(false);
       }
     };
 
-    const debounceTimeout = setTimeout(fetchSuggestions, 100);
+    // Increased debounce time from 100ms to 300ms for better performance
+    const debounceTimeout = setTimeout(fetchSuggestions, 300);
```

Comment:
300 is a bit long. I'm only comfortable raising this to 200 and would prefer to leave it at 100.